src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -14,23 +18,70 @@ namespace Microsoft.AspNet.Mvc.TagHelpers
     /// </summary>
     public class LinkTagHelper : TagHelper
     {
+        private const string HrefIncludeAttributeName = "asp-href-include";
+        private const string HrefExcludeAttributeName = "asp-href-exclude";
         private const string FallbackHrefAttributeName = "asp-fallback-href";
+        private const string FallbackHrefIncludeAttributeName = "asp-fallback-href-include";
+        private const string FallbackHrefExcludeAttributeName = "asp-fallback-href-exclude";
         private const string FallbackTestClassAttributeName = "asp-fallback-test-class";
         private const string FallbackTestPropertyAttributeName = "asp-fallback-test-property";
         private const string FallbackTestValueAttributeName = "asp-fallback-test-value";
         private const string FallbackTestMetaTemplate = "<meta name=\"x-stylesheet-fallback-test\" class=\"{0}\" />";
         private const string FallbackJavaScriptResourceName = "compiler/resources/LinkTagHelper_FallbackJavaScript.js";
 
-        // NOTE: All attributes are required for the LinkTagHelper to process.
-        private static readonly string[] RequiredAttributes = new[]
+        private static readonly Tuple<Mode, string[]>[] ModeInfo =
+            new []
+            {
+                Tuple.Create(Mode.Fallback, new[]
+                {
+                    // Static fallback only
+                    FallbackHrefAttributeName,
+                    FallbackTestClassAttributeName,
+                    FallbackTestPropertyAttributeName,
+                    FallbackTestValueAttributeName
+                }),
+                Tuple.Create(Mode.Fallback, new[]
+                {
+                    // Globbed fallback, include pattern only (static fallback optional)
+                    FallbackHrefIncludeAttributeName,
+                    FallbackTestClassAttributeName,
+                    FallbackTestPropertyAttributeName,
+                    FallbackTestValueAttributeName
+                }),
+                Tuple.Create(Mode.Fallback, new[]
```

Comment:
This case will never get hit/match since the previous one just checks `FallbackHrefIncludeAttributeName` and doesn't require you to even have the `FallbackHrefExcludeAttributeName`. Is it needed?